code/core/src/manager/components/mobile/navigation/useFocusTrap.tsx

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useRef } from 'react';
+
+/**
+ * Hook to trap focus within a container when active
+ *
+ * @template T The type of the HTML element the ref will be attached to.
+ * @param isActive Whether the focus trap should be active
+ * @param onEscape Optional callback to run when Escape key is pressed
+ * @returns A ref to attach to the container element
+ */
+export function useFocusTrap<T extends HTMLElement = HTMLElement>(
+  isActive: boolean,
+  onEscape?: () => void
+): React.RefObject<T> {
+  const containerRef = useRef<T>(null);
+  const previousActiveElementRef = useRef<HTMLElement | null>(null);
+
+  // Store the active element when the trap becomes active and focus the container/first element
+  useEffect(() => {
+    const container = containerRef.current;
+    if (isActive && container) {
+      if (document.activeElement instanceof HTMLElement) {
+        previousActiveElementRef.current = document.activeElement;
+      }
+
+      const interactiveRoleElements = container.querySelectorAll<HTMLElement>(
+        'div[role="button"], span[role="button"], [role="link"]'
+      );
+      interactiveRoleElements.forEach((element) => {
+        if (!element.hasAttribute('tabindex')) {
+          element.setAttribute('tabindex', '0');
+        }
+      });
+
+      const focusableElements = getFocusableElements(container);
```

Comment:
Why is that a different item than the one before? I am not understanding the distinction.